File: Cpp/valid_sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        bool row[9][9] = {false};
        bool col[9][9] = {false};
        bool sqre[9][9] = {false};

        for(size_t i=0; i<9; i++){
            for(size_t j=0; j<9; j++){
                if(board[i][j]=='.') continue;

                int indx = board[i][j] - '0' -1;
                int ar_code = (i/3)*3 + (j/3);

                if(row[i][indx] || col[j][indx]  || sqre[ar_code][indx]){
                    return false;
                }

                row[i][indx] = true;
                col[j][indx] = true;
                sqre[ar_code][indx] = true;
            }
        }
        return true;
    }
};

File: Python/valid_sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        b_temp = [[iq[:3]  for iq in board[:3]],
                [iw[:3]  for iw in board[3:6]],
                [iy[:3] for iy in board[6:9]],
                [ig[3:6] for ig in board[:3]],
                [ih[3:6] for ih in board[3:6]],
                [ij[3:6] for ij in board[6:9]],
                [ik[6:9] for ik in board[:3]], 
                [il[6:9] for il in board[3:6]],
                [im[6:9] for im in board[6:9]]]

        for i in range(9):
            list_1 = board[i]
            list_2 = [item[i] for item in board]
            list_3 = b_temp[i]

            # print(list_1, list_2, list_3, '\n')
            for j in range(1,10):
                empty_list = []
                empty_list.append(list_1.count(str(j)))
                empty_list.append(list_2.count(str(j)))
                empty_list.append(sum(x.count(str(j)) for x in list_3))

                if max(empty_list) > 1:
                    return 0

        return 1